modules/dailymotionBidAdapter.js

```diff
@@ -20,14 +20,20 @@ function getVideoMetadata(bidRequest, bidderRequest) {
   const videoMetadata = {
     description: videoParams.description || '',
     duration: videoParams.duration || 0,
-    iabcat1: deepAccess(bidderRequest, 'ortb2.site.content.cat', [])
+    iabcat1: deepAccess(bidderRequest, 'ortb2.site.content.data', [])
       // Only take IAB cats of taxonomy V1
-      .filter(cat => cat.toLowerCase().indexOf('iab-') === 0),
+      // See https://docs.prebid.org/features/firstPartyData.html#segments-and-taxonomy
+      .filter(data => data?.ext?.segtax === 4)
+      .map(data => (Array.isArray(data?.segment) ? data.segment : []).map(seg => seg.id))
+      .flat(1),
```

Comment:
Is it possible to have the same segment id between the data ?
If it's the case, the final array could contains several times the same value/id.